pkg/providers/tinkerbell/cluster.go

```diff
@@ -108,6 +108,7 @@ func NewClusterSpecValidator(assertions ...ClusterSpecAssertion) *ClusterSpecVal
 		AssertMachineConfigsValid,
 		AssertMachineConfigNamespaceMatchesDatacenterConfig,
 		AssertOsFamilyValid,
+		AssertTinkerbellIPAndControlPlaneIPNotSame,
```

Comment:
Don't you want to also register `AssertTinkerbellIPNotInUse` here?

Reply:
`AssertTinkerbellIPNotInUse` takes in `networkutils.NetClient` as an argument so it doesn't let you register that assertion here. I registered it in `create.go` like how we registered the `NewIPNotInUseAssertion`